packages/vitest/src/node/config/resolveConfig.ts

```diff
@@ -234,26 +234,23 @@ export function resolveConfig(
 
   const browser = resolved.browser
 
-  if (browser.enabled) {
+  // if browser was enabled via CLI and it's configured by the user, then validate the input
+  if (browser.enabled && viteConfig.test?.browser) {
```

Comment:
The main change is that we ignore all browser CLI flags if workspace doesn't have the `browser` configured

Reply:
Note: we throw an error if `--browser` is provided, but there are no configurations for it